website/source/docs/providers/vsphere/r/virtual_machine.html.markdown

```diff
@@ -113,6 +113,7 @@ The `disk` block supports:
 * `template` - (Required if size and bootable_vmdk_path not provided) Template for this disk.
 * `datastore` - (Optional) Datastore for this disk
 * `size` - (Required if template and bootable_vmdks_path not provided) Size of this disk (in GB).
+* `name` - (Required if size is provided) Name of the new disk file to create.
```

Comment:
how about this?

`name` - (Required if size is provided and creating a new disk) This "name" is used as the disk name in vSphere, when the new disk is created.